substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -0,0 +1,246 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.reflect.serialize.hosted;
+
+import com.oracle.svm.core.annotate.AutomaticFeature;
+import com.oracle.svm.core.configure.ConfigurationFiles;
+import com.oracle.svm.core.configure.SerializationConfigurationParser;
+import com.oracle.svm.core.configure.SerializationConfigurationParser.SerializationParserFunction;
+import com.oracle.svm.core.jdk.serialize.SerializationRegistry;
+import com.oracle.svm.core.util.UserError;
+import com.oracle.svm.core.util.VMError;
+import com.oracle.svm.core.util.json.JSONParserException;
+import com.oracle.svm.hosted.FallbackFeature;
+import com.oracle.svm.hosted.FeatureImpl;
+import com.oracle.svm.hosted.ImageClassLoader;
+import com.oracle.svm.hosted.NativeImageOptions;
+import com.oracle.svm.hosted.config.ConfigurationParserUtils;
+import com.oracle.svm.reflect.hosted.ReflectionDataBuilder;
+import com.oracle.svm.reflect.serialize.SerializationSupport;
+import org.graalvm.compiler.options.Option;
+import org.graalvm.compiler.options.OptionKey;
+import org.graalvm.compiler.options.OptionType;
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.impl.RuntimeReflectionSupport;
+
+import java.io.Externalizable;
+import java.io.ObjectStreamField;
+import java.io.Serializable;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.HashSet;
+import java.util.Set;
+
+@AutomaticFeature
+public class SerializationFeature implements Feature {
+    private int loadedConfigurations;
+
+    public static class SerializationFeatureOptions {
+        @Option(help = "Allow user don't set checksum in serialization configuration, and no build time checksum verify will be executed. May lead to unexpected run time behaviors.", type = OptionType.User)
+//
+        public static final OptionKey<Boolean> AllowUncheckedSerialization = new OptionKey<>(false);
+    }
```

Comment:
I'm not sure if we need such an option, we should just support that the checksum is not present in a configuration file, and then not do any checks.

Reply:
Also, this should be a `HostedOptionKey`, then you can simply call `getValue()`.

Reply:
Checksum is used to verify the assumption that the serialization target class' hierarchy keeps consistent through Agent run time, image build time and image run time is held. This assumption is usually held but not guaranteed. Once the assumption is broken, program will run into unexpected behaviors and will be very difficult to debug. The bottom line is we can accurately report the checksum verification failure at runtime so user knows what's going wrong at once. But of cause the earlier we report the problem the better. That's the reason I verify the checksum at both build time and run time.

Users can manually configure serialization item without checksum, but they must acknowledge the risk by explicitly turning on the options.

Reply:
I see, the option is removed. 
If the configured serialization target class doesn't have checksum, a calculated checksum will be assigned to it at build time.

To merge two different checksum classes, currently it will report an error. But it will be fine if one has checksum and another doesn't.

Reply:
Thanks! That merging behavior sounds good.